pyproject.toml

```diff
@@ -8,3 +8,46 @@ readme = "README.md"
 [build-system]
 requires = ["poetry-core"]
 build-backend = "poetry.core.masonry.api"
+
+[tool.poetry.dependencies]
```

Comment:
Is there a way to establish mandatory and optional dependencies?

I am asking because there are dependencies that are only used by specific environments, or specific proxies. So if one just wants to quickly try out the repo with some toy environments, there is absolutely no need to install pymatgen, pyxtal, rdkit, torchani, torch-geometric, torch-scatter, etc.